File: Ahorn/triggers/cassetteTrigger.jl
module FrostHelperCassetteTempoTrigger

using ..Ahorn, Maple

@mapdef Trigger "FrostHelper/CassetteTempoTrigger" FHCassetteTempoTrigger(x::Integer, y::Integer, width::Integer=16, height::Integer=16, Tempo::Number=1.0, ResetOnLeave::Bool=false)


const placements = Ahorn.PlacementDict(
	"Cassette Tempo Trigger (Frost Helper)" => Ahorn.EntityPlacement(
		FHCassetteTempoTrigger,
		"rectangle"
	)
)

end

File: Loenn/triggers/cassetteTrigger.lua
---@module "jautils"
local jautils = require("mods").requireFromPlugin("libraries.jautils")

---@class CassetteTempoTrigger : Trigger
---@field Tempo number
---@field ResetOnLeave boolean

---@type TriggerHandler<CassetteTempoTrigger>
local fHCassetteTempoTrigger = {}
fHCassetteTempoTrigger.name = "FrostHelper/CassetteTempoTrigger"
fHCassetteTempoTrigger.placements = {
    name = "normal",
    data = {
        Tempo = 1.0,
        ResetOnLeave = false,
    }
}

jautils.addExtendedText(fHCassetteTempoTrigger, function (trigger)
    return jautils.roundedToString(trigger.Tempo)
end)

return fHCassetteTempoTrigger